subsys/net/ip/net_if.c

```diff
@@ -108,6 +108,9 @@ static bool net_if_tx(struct net_if *iface)
 #if defined(CONFIG_NET_STATISTICS)
 		pkt_len = net_pkt_get_len(pkt);
 #endif
+#if defined(CONFIG_NET_TCP)
```

Comment:
This commit needs more TLC. The various do_ref_if_needed() calls in tcp.c must to be done **only** if we could actually sent the packet.